src/NHibernate.Test/TypesTest/DecimalTypeFixture.cs

```diff
@@ -22,50 +64,100 @@ protected override string TypeName
 		[Test]
 		public void Equals()
 		{
-			decimal lhs = 5.64351M;
-			decimal rhs = 5.64351M;
+			const decimal lhs = 5.64351M;
+			var rhs = 5.64351M;
 
-			DecimalType type = (DecimalType) NHibernateUtil.Decimal;
-			Assert.IsTrue(type.IsEqual(lhs, rhs));
+			Assert.That(_type.IsEqual(lhs, rhs), Is.True);
 
 			// Test that two decimal fields that are equal except one has a higher precision than
 			// the other one are returned as Equal by the DecimalType.
 			rhs = 5.643510M;
-			Assert.IsTrue(type.IsEqual(lhs, rhs));
+			Assert.That(_type.IsEqual(lhs, rhs), Is.True);
 		}
 
 		[Test]
 		public void ReadWrite()
 		{
-			decimal expected = 5.64351M;
+			const decimal expected = 5.64351M;
 
-			DecimalClass basic = new DecimalClass();
-			basic.Id = 1;
-			basic.DecimalValue = expected;
+			var basic = new DecimalClass
+			{
+				Id = 1,
+				DecimalValue = expected
+			};
 
-			ISession s = OpenSession();
-			s.Save(basic);
-			s.Flush();
-			s.Close();
+			using (var s = OpenSession())
+			{
+				s.Save(basic);
+				s.Flush();
+			}
 
-			s = OpenSession();
-			basic = (DecimalClass) s.Load(typeof(DecimalClass), 1);
+			using (var s = OpenSession())
+			{
+				basic = s.Load<DecimalClass>(1);
 
-			Assert.AreEqual(expected, basic.DecimalValue);
-			Assert.AreEqual(5.643510M, basic.DecimalValue);
-
-			s.Delete(basic);
-			s.Flush();
-			s.Close();
+				Assert.That(basic.DecimalValue, Is.EqualTo(expected));
+				Assert.That(basic.DecimalValue, Is.EqualTo(5.643510M));
+			}
 		}
 
 		[Test]
 		public void UnsavedValue()
 		{
-			DecimalType type = (DecimalType) NHibernateUtil.Decimal;
-			object mappedValue = type.StringToObject("0");
-			Assert.AreEqual(0m, mappedValue);
-			Assert.IsTrue(type.IsEqual(mappedValue, 0m), "'0' in the mapping file should have been converted to a 0m");
+			var mappedValue = _type.StringToObject("0");
+			Assert.That(mappedValue, Is.EqualTo(0m));
+			Assert.IsTrue(_type.IsEqual(mappedValue, 0m), "'0' in the mapping file should have been converted to a 0m");
+		}
+
+		[Test]
+		public void HighScaleParameterSelect()
```

Comment:
Starting with this one, new tests for checking "high" scale decimal are not truncated.